src/engine/javascript_core/engine_value.cc

```diff
@@ -256,6 +281,59 @@ auto Value::to_map() const -> std::map<std::string, Value> {
   return map;
 }
 
+auto Value::to_vector() const -> std::vector<Value> {
+  assert(is_array());
+  JSObjectRef object =
+      get_current_object(this->internal->context, this->internal->value);
+  std::size_t length = get_object_length(this->internal->context, object);
+
+  JSValueRef exception = nullptr;
+  std::vector<Value> vector;
+  for (unsigned int index = 0; index < static_cast<std::size_t>(length);
+       index++) {
+    JSValueRef value = JSObjectGetPropertyAtIndex(this->internal->context,
+                                                  object, index, &exception);
+    assert(!exception);
+    try {
+      vector.emplace_back(static_cast<JSContextRef>(this->internal->context),
+                          value);
+    } catch (const std::exception &) {
+      throw;
+    }
```

Comment:
Is there any need for the try-catch? If not, let's remove it.
```suggestion
    vector.emplace_back(static_cast<JSContextRef>(this->internal->context),
                        value);
```

Reply:
Yeah, you are right.

I think I added it in the first place because I know that some of these std methods could throw.

Example: https://github.com/crossnx/includejs/blob/main/src/engine/javascript_core/engine_value.cc#L245-L252

But in that specific case, I don't prefer that it throws, as I wonder what I could put in the catch lol

An idea?

Reply:
That's correct. Functions in the `std` namespace do throw. But having a `try/catch` that simply rethrows the exception without doing anything special is not different from simply calling the function without the `try/catch`.

Why do you want it to not throw?

Reply:
> That's correct. Functions in the std namespace do throw. But having a try/catch that simply rethrows the exception without doing anything special is not different from simply calling the function without the try/catch.

Yeah I understood this ;) 

> Why do you want it to not throw?

Nothing, I just asked you in case you spot something :)

Reply:
Ah, I think it's fine if it throws :)

Reply:
I mean I put the try/catch, thinking I will fill it later... then I forgot :)

Reply:
(sounds like a bad reflex haha)